miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt

```diff
@@ -21,8 +31,42 @@ internal class MiniAppWebChromeClient(context: Context) : WebChromeClient() {
         super.onReceivedTitle(webView, title)
     }
 
+    // region geolocation
+    private var geoLocationRequestOrigin: String? = null
+    @VisibleForTesting
+    internal var geoLocationCallback: GeolocationPermissions.Callback? = null
+
+    @Suppress("FunctionMaxLength")
+    override fun onGeolocationPermissionsShowPrompt(
```

Comment:
There is use case that miniapps do not call `requestPermission`. They still can request a location access by `navigator.geolocation.getCurrentPosition` -> this will automatically trigger `onGeolocationPermissionsShowPrompt`.

The location access without using the js sdk to `requestPermission` is possible. Our js sample app & client app are working like that (using geolocation).

Reply:
That's true, but I mean the same thing is possible on an Android App: an App could try to use the location without requesting the permission first, and then in that case the request for the location would throw an error. 

When a Mini App requests the location, they pass in an error listener `navigator.geolocation.getCurrentPosition(pos => console.log(pos), e => console.error(e))`. So in the case that they didn't called `requestPermission` first, then the browser will call the error listener.

Also, I think it's better if we can leave all of the device permission requesting out of the SDK and leave it up to the Host App to handle.

Reply:
Okay, I see your point now.
I'll remove the permission handle in `onGeolocationPermissionsShowPrompt` so the MiniApps can only request a permission via js sdk.

Reply:
+1 -> It's better if we can leave all of the device permission requesting out of the SDK and leave it up to the Host App to handle.